lib/src/asset_graph/exceptions.dart

```diff
@@ -11,3 +11,14 @@ class DuplicateAssetNodeException implements Exception {
   @override
   String toString() => 'DuplicateAssetNodeError: $assetNode';
 }
+
+class AssetGraphVersionError implements Exception {
+  final int versionSeen;
+  final int currentVersion;
+
+  AssetGraphVersionError(this.versionSeen, this.currentVersion);
+
+  @override
+  String toString() => 'AssetGraphVersionError: saw version $versionSeen but '
```

Comment:
Actually I need to rename this to `Exception` not `Error`. In practice I plan on catching this Exception and just starting the build from scratch in this case, so the user should never see the message.